classes/DataWarehouse/Visualization/AppKernelChart.php

```diff
@@ -0,0 +1,1075 @@
+<?php
+
+namespace DataWarehouse\Visualization;
+
+/**
+ * Base class for server generated chart for app kernel data.
+ *
+ * @author Amin Ghadersohi
+ */
+class AppKernelChart extends AggregateChart
+{
+    const IN_CONTROL_COLOR = 'rgba(170, 255, 170, 0.75)';
+
+    const CONTROL_REGION_TIME_INTERVAL_COLOR = '#E0F8F7';
+
+    const OVERPERFORMING_COLOR = '#F2F5A9';
+
+    const UNDERPERFORMING_COLOR = '#F5A9BC';
+
+    const MINMAX_COLOR = '#cc99ff';
+
+    public $_axis_index;
+
+    /**
+     * CSS url containing Base 64 encoded indicator image.
+     *
+     * @var string
+     */
+    protected $_indicator_url;
```

Comment:
So the only comment I had was it looks like in the original `HighChartAppKernel.php` it actually set this to the base64 encoded contents of an image while in this class it's set to the url of the image. Was that needed because of Plotly?

Reply:
Plotly [images](https://plotly.com/javascript/reference/layout/images/#layout-images-items-image-source) allows for absolute or relative URLs passed in as the image source. I think I remember I was having a hard time getting it to accept the base64 encoded contents so I thought it would be more straightforward to use the URL instead.

Reply:
*nod* sounds good.